packages/story-editor/src/components/colorPicker/basicColorPicker.js

```diff
@@ -76,58 +100,134 @@ function BasicColorPicker({
-          {__('Default colors', 'web-stories')}
-        </DefaultText>
+      <Header
+        handleClose={handleClose}
+        isEditMode={isEditMode}
+        setIsEditMode={setIsEditMode}
+        hasPresets={hasPresets}
+      >
+        <DefaultText>{__('Color', 'web-stories')}</DefaultText>
       </Header>
       <Body>
-        <BasicColorList
-          color={color}
-          colors={BASIC_COLORS}
-          handleColorChange={handleColorChange}
-          allowsOpacity={allowsOpacity}
-          allowsGradient={allowsGradient}
-          aria-labelledby="colorpicker-default-colors-title"
-        />
-        {allowsSavedColors && savedColors.length > 0 && (
-          <>
-            <Label id="colorpicker-saved-colors-title">
-              {__('Saved colors', 'web-stories')}
-            </Label>
-            <BasicColorList
-              color={color}
-              colors={savedColors}
-              handleColorChange={handleColorChange}
-              allowsOpacity={allowsOpacity}
-              allowsGradient={allowsGradient}
-              aria-labelledby="colorpicker-saved-colors-title"
-            />
-          </>
-        )}
-        {allowsSavedColors && storyColors.length > 0 && (
-          <>
-            <Label id="colorpicker-story-colors-title">
-              {__('Story colors', 'web-stories')}
-            </Label>
-            <BasicColorList
-              color={color}
-              colors={storyColors}
-              handleColorChange={handleColorChange}
-              allowsOpacity={allowsOpacity}
-              allowsGradient={allowsGradient}
-              aria-labelledby="colorpicker-story-colors-title"
-            />
-          </>
-        )}
+        <EyedropperWrapper>
+          <Button
+            variant={BUTTON_VARIANTS.SQUARE}
+            type={BUTTON_TYPES.QUATERNARY}
+            size={BUTTON_SIZES.SMALL}
+            aria-label={__('Pick a color from canvas', 'web-stories')}
+            onClick={initEyedropper()}
+            onPointerEnter={initEyedropper(false)}
+          >
+            <Icons.Pipette />
+          </Button>
+          <StyledText size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL}>
+            {__('Sample color', 'web-stories')}
+          </StyledText>
+        </EyedropperWrapper>
+        <SavedColors>
+          {allowsSavedColors && (
+            <>
+              <Label id="colorpicker-story-colors-title">
+                {__('Current story', 'web-stories')}
+              </Label>
+              <BasicColorList
+                color={color}
+                colors={storyColors}
+                handleClick={handleClick}
+                colorType="local"
+                allowsOpacity={allowsOpacity}
+                allowsGradient={allowsGradient}
+                aria-labelledby="colorpicker-story-colors-title"
+                isEditMode={isEditMode}
+              />
+            </>
+          )}
+          {allowsSavedColors && (
+            <>
```

Comment:
Couldn't these four lines be removed? It's the same condition guarding the contents, no?